src/browser/ui/ReactDOMComponent.js

```diff
@@ -234,8 +247,13 @@ ReactDOMComponent.Mixin = {
           }
           propValue = CSSPropertyOperations.createMarkupForStyles(propValue);
         }
-        var markup =
-          DOMPropertyOperations.createMarkupForProperty(propKey, propValue);
+        var markup = null;
+        if (this._tag != null && this._tag.indexOf('-') >= 0) {
+          validateDangerousAttributeName(propKey);
+          markup = DOMPropertyOperations.createMarkupForCustomAttribute(propKey, propValue);
```

Comment:
I think that's how I originally had it written up, but it resulted in a couple of things that Paul or Sebastian didn't like.  There is a remnant of the old implementation in the jsdocs over createMarkupForCustomAttribute (jsdocs specify a tagName parameter that was since removed); I'll go clean this up when we come to a decision here.  I don't remember the details, and I'm happy to change it back, but maybe get one of them to say yay/nay before we make a change that they don't want to accept.

I *think* the issues might have been one or more of the following:
* DOMPropertyOperations now has a cache, and they didn't want caching at that layer?  Someone wanted that layer to be a pure function of its inputs?
* It leads to an awkward set of rules about who is responsible for validation.  For non-custom-attributes, they safety is verified by the whitelist, but for custom attributes we would be verifying in DOMPropertyOperations.  Now you can't say "DOMPropertyOperations is unsafe and ReactDOMComponent is safe"... the statement becomes "things are safe because of a complicated interweaving of security checks."
* Some other reason that isn't on this list.

Anyway, my gut intuition is that validation should be done in DOMPropetyOperations (so I agree with you on that point), but I can also understand the other side of the arguments.

CC: @zpao @sebmarkbage , can we get one of you to weigh in on this topic.  Where should each of these validations occur?